packages/utils/src/configured_bignumber.ts

```diff
@@ -4,6 +4,7 @@ import { BigNumber } from 'bignumber.js';
 // more then 20 digits. We want to avoid this behavior, so we set EXPONENTIAL_AT to a high number
 BigNumber.config({
     EXPONENTIAL_AT: 1000,
+    DECIMAL_PLACES: 100,
```

Comment:
I was surprised to find that `bignumber.js` *does not* support infinite precision division. (I'm used to BigNumber libraries that will treat results of division as rational numbers consisting of a numerator and denominator when possible, thereby allowing theoretically infinite precision). Some tests were failing because `(reallyBigNumber.sub(1)) / (reallyBigNumber) === new BigNumber(1)` for sufficiently large values of `reallyBigNumber`. This change helps, but it is worth noting that it is *still* possible for rounding issues when dividing really big numbers.

Reply:
BigNumber.js is a float-point library with arbitrary precision. We should stay in the domain of integers, which means not using `div` but always using integer division with `divToInt`. Sometimes it requires a bit of algebra to create the correct verification logic in tests. (i.e. don't test A/B = C, but test A = B * C).

Reply:
For 512 bits of precision (product of two 256-bit numbers) we would need at least 155 decimals. Depending on how we build our tests, we might need more.

But like mentioned, I would rather stay in the exact domain and avoid rounding everywhere. According to the BigNumber.js docs only division and related operations such as square root, base conversion and negative powers involve rounding. It seems`divToInt` does not use `DECIMAL_PLACES` and is always precise.

Perhaps we can set `DECIMAL_PLACES` to a perversely low value to make tests fail that use imprecise operations.

Reply:
I discovered `divToInt` after making this change and I am using now in the one place in our tests where we need to do division. So the immediate issue is solved, but it still kind of scares me that it's so easy to produce precision-related errors.

@fabioberger any thoughts here regarding how it affects our users?